public/app/features/query/state/runRequest.ts

```diff
@@ -186,6 +186,20 @@ export function runRequest(
   return merge(timer(200).pipe(mapTo(state.panelData), takeUntil(dataObservable)), dataObservable);
 }
 
+function callQueryMethodWithMigration(
+  datasource: DataSourceApi | DataSourceWithBackendMigration,
+  request: DataQueryRequest,
+  queryFunction?: typeof datasource.query
+) {
+  if (datasource instanceof DataSourceWithBackendMigration) {
```

Comment:
If the _only_ purpose of this class is to indicate that the data source supports migrations, we could consider other ways of doing that. A flag in plugin.json perhaps?

Reply:
moved to a interface, as discussed offline.